go/runtime/scheduling/simple/txpool/api/api.go

```diff
@@ -33,8 +33,20 @@ type TxPool interface {
 	// GetBatch gets a transaction batch from the transaction pool.
 	GetBatch(force bool) []*transaction.CheckedTransaction
 
+	// GetKnownBatch gets a set of known transactions from the transaction pool.
+	//
+	// For any missing transactions nil will be returned in their place and the map of missing
+	// transactions will be populated accoordingly.
+	GetKnownBatch(batch []hash.Hash) ([]*transaction.CheckedTransaction, map[hash.Hash]int)
+
+	// GetTransactions returns the given number of transactions from the transaction pool without
+	// taking any batch limits into account.
```

Comment:
Maybe mention that this also ignores priority (weights), right?

Reply:
Good idea, added.